pkg/providers/merge/event.go

```diff
@@ -104,17 +106,60 @@ func (m *merge) ProcessIssueCommentEvent(event *github.IssueCommentEvent) {
 		}
 		util.Error(m.addCanMerge(pr))
 		model := AutoMerge{
-			PrID:      pr.GetNumber(),
-			Owner:     m.owner,
-			Repo:      m.repo,
-			BaseRef:   pr.GetBase().GetRef(),
-			Status:    false,
-			CreatedAt: time.Now(),
+			PrID:         pr.GetNumber(),
+			Owner:        m.owner,
+			Repo:         m.repo,
+			BaseRef:      pr.GetBase().GetRef(),
+			Status:       false,
+			CreatedAt:    time.Now(),
+			WithoutTests: body == withoutTestsMergeCommand,
 		}
+
+		if model.WithoutTests {
+			lastRunAllTestTime, err := m.getLastRunAllTestAt(pr)
+			if err != nil {
+				// When we can not find the last run all tests command, we need to set the without tests flag as false.
+				model.WithoutTests = false
+				util.Error(errors.Wrap(err, "get last run all test time"))
+			} else {
+				model.LastTestAllAt = lastRunAllTestTime
+			}
+		}
+
 		if err := m.saveModel(&model); err != nil {
 			util.Error(errors.Wrap(err, "merge process PR event"))
 		} else {
 			util.Error(m.queueComment(pr))
 		}
 	}
 }
+
+// getLastRunAllTestAt get last run all test comment time.
+func (m *merge) getLastRunAllTestAt(pr *github.PullRequest) (time.Time, error) {
+	if pr == nil {
+		return time.Time{}, errors.New("got a nil pr")
+	}
+
+	// Get comments in descending order.
+	options := &github.PullRequestListCommentsOptions{
+		Direction: "desc",
+	}
+
+	comments, _, err := m.opr.Github.PullRequests.ListComments(context.Background(), m.owner, m.repo,
+		pr.GetNumber(), options)
```

Comment:
Should use `m.opr.Github.Issues.ListComments` here.

- issue comments: https://api.github.com/repos/pingcap-incubator/cherry-bot/issues/43/comments
- pull comments: https://api.github.com/repos/pingcap-incubator/cherry-bot/pulls/43/comments